frame/bags-list/src/tests.rs

```diff
@@ -196,6 +196,284 @@ mod pallet {
 			assert_storage_noop!(assert!(BagsList::rebag(Origin::signed(0), 10).is_ok()));
 		})
 	}
+
+	#[test]
+	fn put_in_front_of_two_node_bag_heavier_is_tail() {
+		ExtBuilder::default().add_ids(vec![(10, 15), (11, 16)]).build_and_execute(|| {
+			// given
+			assert_eq!(
+				List::<Runtime>::get_bags(),
+				vec![(10, vec![1]), (20, vec![10, 11]), (1_000, vec![2, 3, 4])]
+			);
+
+			StakingMock::set_vote_weight_of(&10, 15);
+			StakingMock::set_vote_weight_of(&11, 16);
+
+			// when
+			assert_ok!(BagsList::put_in_front_of(Origin::signed(11), 10));
+
+			// then
+			assert_eq!(
+				List::<Runtime>::get_bags(),
+				vec![(10, vec![1]), (20, vec![11, 10]), (1_000, vec![2, 3, 4])]
+			);
+		});
+	}
+
+	#[test]
+	fn put_in_front_of_two_node_bag_heavier_is_head() {
+		ExtBuilder::default().add_ids(vec![(11, 16), (10, 15)]).build_and_execute(|| {
+			// given
+			assert_eq!(
+				List::<Runtime>::get_bags(),
+				vec![(10, vec![1]), (20, vec![11, 10]), (1_000, vec![2, 3, 4])]
+			);
+
+			StakingMock::set_vote_weight_of(&11, 16);
+			StakingMock::set_vote_weight_of(&10, 15);
+
+			// when
+			assert_ok!(BagsList::put_in_front_of(Origin::signed(11), 10));
+
+			// then
+			assert_eq!(
+				List::<Runtime>::get_bags(),
+				vec![(10, vec![1]), (20, vec![11, 10]), (1_000, vec![2, 3, 4])]
+			);
+		});
+	}
+
+	#[test]
+	fn put_in_front_of_non_terminal_nodes_heavier_behind() {
+		ExtBuilder::default().add_ids(vec![(5, 1_000)]).build_and_execute(|| {
+			// given
+			assert_eq!(List::<Runtime>::get_bags(), vec![(10, vec![1]), (1_000, vec![2, 3, 4, 5])]);
+
+			StakingMock::set_vote_weight_of(&4, 1_000);
+			StakingMock::set_vote_weight_of(&3, 999);
```

Comment:
this is an interesting test, because 3 actually needs a rebag as well! perhaps the test can be written differently to avoid this.

As for the rebag, I don't think it is an issue. We compare against the real weight of all nodes, so everything should be correct. As an example, if someone has a lot of vote weight, but is mistakenly in a small bag, you cannot overtake the, which is totally fair

Reply:
EDIT: sorry, it does not a rebag actually, but my second comment holds anyway.